src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1244,20 +1246,7 @@ private void requestTool(@NotNull final BlockState target, final BlockPos pos)
     }
 
     public boolean canUseShears() {
-        if (building instanceof BuildingBuilder)
-        {
-            return building.getSetting(BuildingBuilder.USE_SHEARS).getValue();
-        }
-        else if (building instanceof BuildingLumberjack)
-        {
-            return building.getSetting(BuildingLumberjack.USE_SHEARS).getValue();
-        }
-        else if (building instanceof BuildingMiner)
-        {
-            return building.getSetting(BuildingMiner.USE_SHEARS).getValue();
-        }
-
-        return false;
+        return building.getOptionalSetting(USE_SHEARS).orElse(new BoolSetting(false)).getValue();
```

Comment:
move this call into the method that checks for the tools and remove canUseShears method entirely then

Reply:
There are 2 methods that need this: getMostEfficientTool and requestTool

Reply:
You can either hand the AI that will handle it, or the building, and then query it at the place you need it. 
The canUseShears should not be a method param.